### ISSUE TITLE
(RedM/new natives check): unkown userdata pointer

### ISSUE THREAD

**Issue by d22tny** [citizenfx/fivem#2890]
### What happened?

As i think the new natives check got merged into the release build, players on my server started sending me photos of console errors as things are not working. But i have no idea what `unkown userdata pointer` means.

![image](https://github.com/user-attachments/assets/b78fe77c-ee36-4e21-ba50-162107de1afe)

RemoveBlip(self.blip) that's the code it's erroring. /
DeleteEntity(ped) /
SetVehicleAsNoLongerNeeded(veh).

The only thing that i see wrong about the code itself it's that it doesn't check if the blip / entity really exists, but seeing from the player's perspective, it does exist.

### Expected result

-

### Reproduction steps

Can't be really reproed since it happens randomly after some play time.

### Importancy

Slight inconvenience

### Area(s)

FiveM, RedM

### Specific version(s)

RedM, 10653 sv build

**Comment by outsider31000** [comment]
so far I havent received any reports with issues like this , but yeah those errors are looking weird
but what you mean by "new natives check" ?

**Comment by d22tny** [comment]
https://github.com/citizenfx/fivem/commit/e7234c42eb74c0835801436a8ad64c498af23a8f

**Comment by outsider31000** [comment]
local GunHash = `WEAPON_RIFLE_SPRINGFIELD`
   SetCurrentPedWeapon(GunHash, 0, 0)
   does this native errors for you too ?

**Comment by outsider31000** [comment]
just updated artifacts and get errors on the older it was fine

**Comment by outsider31000** [comment]
![image](https://github.com/user-attachments/assets/a9cc064f-9bc7-46a8-a261-e37734444764)

**Comment by d22tny** [comment]
> ```lua
> local GunHash = `WEAPON_RIFLE_SPRINGFIELD`
> SetCurrentPedWeapon(GunHash, 0, 0) -- note that the native has wrong parameteres player ped should be where gun hash is on older artifact it wasnt erroring
> ```
> 
> does this native errors for you too ? it errors if i add wrong parameter, but wasnt doing that in older artifacts
> 
> ![381312127-a9cc064f-9bc7-46a8-a261-e37734444764](https://private-user-images.githubusercontent.com/87246847/381313478-e5ca2e30-8de3-44be-8f34-843cc21d3440.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyNDAyODQsIm5iZiI6MTczMDIzOTk4NCwicGF0aCI6Ii84NzI0Njg0Ny8zODEzMTM0NzgtZTVjYTJlMzAtOGRlMy00NGJlLThmMzQtODQzY2MyMWQzNDQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI5VDIyMTMwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ2MjIzNjUyZDVmYzI0YTFlOTJhNDE4NmI1YzYxOWI0YmE4ZjQzNmExNDdmM2YzMmZlMjVjMTUyZTg0YmQ2YTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZAwQRjIpFx-WID85Rr9krRGghDh5M5pUKAoZS8SqdFU)

this works kinda as expected, it should error out if wrong parameters are sent, but it doesn't display the actual error info.

**Comment by outsider31000** [comment]
yes but in older artifact it doesnt error

**Comment by outsider31000** [comment]
literally tested 5 minutos before switching to the new artifact because theres an issue where players can't hear the last bullet from weapons and didnt noticed the params, when updated, then it errored and I saw why.

**Comment by iridium-cfx** [comment]
Are you calling `Citizen.PointerValue[Int/Float]Initialized` outside of the natives, or are you getting any other native errors before this one?
Currently the values returned by that function are stored in a pool (and released when calling the native with them), but if the pool runs out it will return a value which would explain the error you are getting

**Comment by d22tny** [comment]
Yes, @iridium-cfx . On the SetVehicleAsNoLongerNeeded, i call that :

`SetVehicleAsNoLongerNeeded(Citizen.PointerValueIntInitialized(veh))`

But on the other two errors, though, i don't.

DeleteEntity error:

```
if self.pedList[k] then
    DeleteEntity(self.pedList[k].entity)
    self.pedList[k] = nil
end
```

RemoveBlip error:

`RemoveBlip(self.blip)`

**Comment by iridium-cfx** [comment]
> Yes, @iridium-cfx . On the SetVehicleAsNoLongerNeeded, i call that :
> 
> `SetVehicleAsNoLongerNeeded(Citizen.PointerValueIntInitialized(veh))`
> 
> But on the other two errors, though, i don't.
> 
> DeleteEntity error:
> 
> ```
> if self.pedList[k] then
>     DeleteEntity(self.pedList[k].entity)
>     self.pedList[k] = nil
> end
> ```
> 
> RemoveBlip error:
> 
> `RemoveBlip(self.blip)`
> 
> Other native errors are not present before those.

Ah, right. `SetVehicleAsNoLongerNeeded` already calls PointerValueIntInitialized internally You shouldn't be calling it yourself.
```lua
-- _ii = Citizen.PointerValueIntInitialized`
function Global.SetVehicleAsNoLongerNeeded(vehicle)
	return _in(0x629BFA74418D6239, _ii(vehicle) --[[ may be optional ]])
end
```

**Comment by d22tny** [comment]
Okay, so it was a chain effect? The pool broke because of the first error and errored the RemoveBlip and DeleteEntity because of the first error, right ?

**Comment by iridium-cfx** [comment]
> Okay, so it was a chain effect? The pool broke because of the first error and errored the RemoveBlip and DeleteEntity because of the first error, right ?

I'd guess so, yeah. Once it's broke you are kinda screwed until restart.
I'm going to to tweak this code anyway, but hopefully removing your own call to `Citizen.PointerValueIntInitialized` should fix the issue.

**Comment by d22tny** [comment]
Definetly, after your explanation i understand how it works. I'll check them all and i'll fix them. Thank you a lot! If you need additional info from me in order to research, i can gladly help, otherwise you can close the issue! Thanks again.

**Comment by d22tny** [comment]
So i've made my research ( if anyone else gets into this problem ), these natives ( which require initialized pointers ) already have it called so you don't need to call it yourself. :

```
'DELETE_ENTITY',
'DELETE_OBJECT',
'DELETE_PED',
'DELETE_VEHICLE',
'DELETE_MISSION_TRAIN',
'SET_ENTITY_AS_NO_LONGER_NEEDED',
'SET_OBJECT_AS_NO_LONGER_NEEDED',
'SET_PED_AS_NO_LONGER_NEEDED',
'SET_VEHICLE_AS_NO_LONGER_NEEDED'
```
